examples/build-package/main.tf

```diff
@@ -17,28 +17,52 @@ resource "random_pet" "this" {
 # Build packages
 #################
 
-# Create zip-archive of a single directory where "pip install" will also be executed (default for python runtime)
+# Create zip-archive of a single directory where "pip install" will also be executed (default for python runtime with requirements.txt present)
 module "package_dir" {
   source = "../../"
 
   create_function = false
 
-  runtime     = "python3.8"
-  source_path = "${path.module}/../fixtures/python3.8-app1"
+  build_in_docker = true
+  runtime         = "python3.8"
+  source_path     = "${path.module}/../fixtures/python3.8-app1"
+  artifacts_dir   = "${path.root}/builds/package_dir/"
 }
 
-# Create zip-archive of a single directory where "pip install" will also be executed (default for python runtime) and set temporary directory for pip install
+# Create zip-archive of a single directory where "pip install" will also be executed (default for python runtime with requirements.txt present) and set temporary directory for pip install
 module "package_dir_pip_dir" {
   source = "../../"
 
   create_function = false
 
-  runtime = "python3.8"
+  build_in_docker = true
+  runtime         = "python3.8"
   source_path = [{
     path             = "${path.module}/../fixtures/python3.8-app1"
     pip_tmp_dir      = "${path.cwd}/../fixtures"
     pip_requirements = "${path.module}/../fixtures/python3.8-app1/requirements.txt"
   }]
+  artifacts_dir = "${path.root}/builds/package_dir_pip_dir/"
+}
+
+# Create zip-archive of a single directory where "poetry export" & "pip install --no-deps" will also be executed
+module "package_dir_poetry" {
+  source = "../../"
+
+  create_function = false
+
+  build_in_docker = true
+  runtime         = "python3.8"
+  docker_image    = "build-python3.8-poetry"
+  docker_file     = "${path.module}/../fixtures/python3.8-app-poetry/docker/Dockerfile"
+
+  source_path = [
+    {
+      path           = "${path.module}/../fixtures/python3.8-app-poetry"
+      poetry_install = true
```

Comment:
Maybe rename this to be more explicit that `poetry export` then `pip install --no-deps` are executed.